pytest_factoryboy/fixture.py

```diff
@@ -99,41 +154,55 @@ def register(factory_class, _name=None, **kwargs):
                 if isinstance(value, factory.SubFactory):
                     args.append(inflection.underscore(subfactory_class._meta.model.__name__))
 
-                make_fixture(
-                    name=attr_name,
-                    module=module,
-                    func=subfactory_fixture,
-                    args=args,
-                    factory_class=subfactory_class,
+                fixture_defs.append(
+                    FixtureDef(
+                        name=attr_name,
+                        function_name="subfactory_fixture",
+                        function_kwargs={"factory_class": subfactory_class},
+                        deps=args,
+                    )
                 )
             else:
                 if isinstance(value, LazyFixture):
                     args = value.args
 
-                make_fixture(
-                    name=attr_name,
-                    module=module,
-                    func=attr_fixture,
-                    value=value,
-                    args=args,
+                fixture_defs.append(
+                    FixtureDef(
+                        name=attr_name,
+                        function_name="attr_fixture",
+                        function_kwargs={"value": value},
+                        deps=args,
+                    )
                 )
 
     if not hasattr(module, factory_name):
-        make_fixture(
-            name=factory_name,
-            module=module,
-            func=factory_fixture,
-            factory_class=factory_class,
+        fixture_defs.append(
+            FixtureDef(
+                name=factory_name,
+                function_name="factory_fixture",
+                function_kwargs={"factory_class": factory_class},
+            )
         )
 
-    make_fixture(
-        name=model_name,
-        module=module,
-        func=model_fixture,
-        args=deps,
-        factory_name=factory_name,
-        related=related,
+    fixture_defs.append(
+        FixtureDef(
+            name=model_name,
+            function_name="model_fixture",
+            function_kwargs={"factory_name": factory_name},
+            deps=deps,
+            related=related,
+        )
     )
+
+    code = module_template.render(fixture_defs=fixture_defs)
+    mod = make_module(code, module_name=model_name, package_name="_pytest_factoryboy_generated_fixtures")
```

Comment:
NIP: what do you think about using some more meaningful name e.g. `generated_module` or `model_fixtures_module` etc?